src/utilities/createRouter.spec.ts

```diff
@@ -86,6 +92,23 @@ test('route is readonly except for individual params', async () => {
 
   // @ts-expect-error
   route.params.param = 'goodbye'
+
+  expect(spy).toHaveBeenCalledOnce()
+
+  await flushPromises()
+
   // @ts-expect-error
   expect(route.params.param).toBe('goodbye')
+
+  // @ts-expect-error
```

Comment:
is this expect-error here because we actually expect an error or because types weren't setup correctly yet?

Reply:
Right it's because params is type {} which means it has no properties so you cannot access anything. The two you flagged should go away once route.params is Record<string, unknown>